### ISSUE TITLE
st2 run --tail stops tailing workflows after first sub-workflow completes

### ISSUE THREAD

**Issue by nmaludy** [StackStorm/st2#3960]
When tailing a Mistral workflow that has tasks that are also Mistral workflows (mistral-v2 actions) the tailing stops when the first sub-workflow completes instead of waiting for the root workflow to complete.

Example:
```shell
[root@stackstorm ~]# st2 run --tail examples.mistral-workbook-subflows
Tailing execution "5a67e8f9a814c0349e2fa1f9"
Child execution (task=print_subject) 5a67e8f9a814c0349e2fa1fc has started.

Child execution (task=print_subject) 5a67e8f9a814c0349e2fa1fc has started.

5a67e8f9a814c0349e2fa1fc

Execution 5a67e8f9a814c0349e2fa1f9 has completed (status=succeeded).

id: 5a67e8f9a814c0349e2fa1f9
action.ref: examples.mistral-workbook-subflows
parameters: None
status: running
start_timestamp: Wed, 24 Jan 2018 02:01:29 UTC
end_timestamp: 
+-----------------------------+------------------------+---------------+--------------------+-----------------+
| id                          | status                 | task          | action             | start_timestamp |
+-----------------------------+------------------------+---------------+--------------------+-----------------+
| + 5a67e8f9a814c0349e2fa1fc  | running (1s elapsed)   | print_subject | examples.mistral-  | Wed, 24 Jan     |
|                             |                        |               | basic              | 2018 02:01:29   |
|                             |                        |               |                    | UTC             |
|    5a67e8faa814c0349e2fa1fe | succeeded (0s elapsed) | task1         | core.local         | Wed, 24 Jan     |
|                             |                        |               |                    | 2018 02:01:30   |
|                             |                        |               |                    | UTC             |
+-----------------------------+------------------------+---------------+--------------------+-----------------+
```

You can see this stops after the first task `print_subject` action  `examples.mistral-basic` completes.

Looking at the full output afterwards you can see there is another action (workflow) that is executed after.

``` shell
[root@stackstorm ~]# st2 execution get 5a67e8f9a814c0349e2fa1f9
id: 5a67e8f9a814c0349e2fa1f9
action.ref: examples.mistral-workbook-subflows
parameters: None
status: succeeded (7s elapsed)
result_task: print_adjective
result: 
  extra:
    state: SUCCESS
    state_info: null
  stdout: 5a67e8fca814c0349e2fa200
  tasks:
  - created_at: '2018-01-24 02:01:32'
    id: b02a0230-5283-4945-a830-fcb730abf105
    input: null
    name: task1
    published:
      stdout: 5a67e8fca814c0349e2fa200
    result:
      failed: false
      return_code: 0
      stderr: ''
      stdout: 5a67e8fca814c0349e2fa200
      succeeded: true
    state: SUCCESS
    state_info: null
    updated_at: '2018-01-24 02:01:33'
    workflow_execution_id: 8c9fcedf-404b-4ca9-ab2f-a25a24d0e06e
    workflow_name: examples.mistral-basic
start_timestamp: Wed, 24 Jan 2018 02:01:29 UTC
end_timestamp: Wed, 24 Jan 2018 02:01:36 UTC
+-----------------------------+------------------------+-----------------+--------------------+-----------------+
| id                          | status                 | task            | action             | start_timestamp |
+-----------------------------+------------------------+-----------------+--------------------+-----------------+
| + 5a67e8f9a814c0349e2fa1fc  | succeeded (3s elapsed) | print_subject   | examples.mistral-  | Wed, 24 Jan     |
|                             |                        |                 | basic              | 2018 02:01:29   |
|                             |                        |                 |                    | UTC             |
|    5a67e8faa814c0349e2fa1fe | succeeded (0s elapsed) | task1           | core.local         | Wed, 24 Jan     |
|                             |                        |                 |                    | 2018 02:01:30   |
|                             |                        |                 |                    | UTC             |
| + 5a67e8fca814c0349e2fa200  | succeeded (2s elapsed) | print_adjective | examples.mistral-  | Wed, 24 Jan     |
|                             |                        |                 | basic              | 2018 02:01:32   |
|                             |                        |                 |                    | UTC             |
|    5a67e8fca814c0349e2fa202 | succeeded (1s elapsed) | task1           | core.local         | Wed, 24 Jan     |
|                             |                        |                 |                    | 2018 02:01:32   |
|                             |                        |                 |                    | UTC             |
+-----------------------------+------------------------+-----------------+--------------------+-----------------+
```

**Comment by Kami** [comment]
Thanks for reporting this.

Do you have a simple sample workflow which reproduces this problem?

In the past I tested it with action chain and it should work fine for single level nested workflows (but not more), but there could also be some edge case which is not handled correctly.

The CLI code should be workflow engine independent, since it just uses execution ids, but could be something else going on.

**Comment by nmaludy** [comment]
@Kami the following action in the examples pack reproduces this `examples.mistral-workbook-subflows`
https://github.com/StackStorm/st2/blob/master/contrib/examples/actions/mistral-workbook-subflows.yaml

**Comment by Kami** [comment]
Thanks.

I'm looking into it and yeah, it looks like it's indent related to known / documented limitation of only supporting single level nested executions (in this case it's workflow -> workflow -> execution, but only workflow -> execution is supported right now).

**Comment by Kami** [comment]
I will look into a small change to also support two levels of nested executions (such as in this case, workflow -> workflow -> execution), but supporting arbitrary nested executions will require more work in the future.

**Comment by nmaludy** [comment]
@Kami that's great, but at some point I would really love arbitrarily deeply nested support. We have tons of deeply nested workflows and being able to tail their progress would be awesome considering we almost exclusively work out of the CLI.

**Comment by Kami** [comment]
@nmaludy I this case feel free to come up with a proposal of how the output should look like in such scenarios.

Imo, it's simply too hard to track multi level nested executions in the CLI without using some kind of tree output or similar, but even then get's messy and unfriendly.

**Comment by Kami** [comment]
I've merged #3962 for now.

If you have an idea of how to display arbitrarily nested executions in a manner which is obvious and makes sense (aka so it's clear something is a child task, to which parent workflow it belongs, etc.), please open an issue with the proposal
